packages/components/link/src/link.tsx

```diff
@@ -0,0 +1,143 @@
+import type { Theme } from '@emotion/react';
+import type { MessageDescriptor } from 'react-intl';
+import React from 'react';
+import { Link as ReactRouterLink } from 'react-router-dom';
+import { css, useTheme } from '@emotion/react';
+import { FormattedMessage } from 'react-intl';
+import { customProperties as vars } from '@commercetools-uikit/design-system';
+import { filterInvalidAttributes, warning } from '@commercetools-uikit/utils';
+
+type TExtendedTheme = Theme & {
+  [key: string]: string;
+};
+type TLinkShape = {
+  pathname: string;
+  search: string;
+  hash?: string;
+  state: unknown;
+};
```

Comment:
I'm tempted to use the `LocationDescriptor` instead, because it makes a lot more sense than us having to define it ourselves

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/04657b253bcdd391b28709cbbf9bd8d78f81611c/types/react-router/v2/lib/Router.d.ts#L36:L41

This will mean dep on `@types/history`.

what do you think?